twoliter/src/tools.rs

```diff
@@ -16,36 +16,38 @@ const PUBSYS_SETUP: &[u8] = include_bytes!(env!("CARGO_BIN_FILE_PUBSYS_SETUP"));
 const TESTSYS: &[u8] = include_bytes!(env!("CARGO_BIN_FILE_TESTSYS"));
 const TUFTOOL: &[u8] = include_bytes!(env!("CARGO_BIN_FILE_TUFTOOL"));
 
-/// Create a `TempDir` object and provide a tools-centric error message if it fails. Make sure you
-/// hang on to the `TempDir` for as long as you need it. It will be deleted when it goes out of
-/// scope.
-pub(crate) fn tools_tempdir() -> Result<TempDir> {
-    TempDir::new().context("Unable to create a tempdir for Twoliter's tools")
-}
-
 /// Install tools into the given `tools_dir`. If you use a `TempDir` object, make sure to pass it by
 /// reference and hold on to it until you no longer need the tools to still be installed (it will
 /// auto delete when it goes out of scope).
 pub(crate) async fn install_tools(tools_dir: impl AsRef<Path>) -> Result<()> {
     let dir = tools_dir.as_ref();
     debug!("Installing tools to '{}'", dir.display());
 
-    write_bin("bottlerocket-variant", BOTTLEROCKET_VARIANT, &dir).await?;
-    write_bin("buildsys", BUILDSYS, &dir).await?;
-    write_bin("pubsys", PUBSYS, &dir).await?;
-    write_bin("pubsys-setup", PUBSYS_SETUP, &dir).await?;
-    write_bin("testsys", TESTSYS, &dir).await?;
-    write_bin("tuftool", TUFTOOL, &dir).await?;
-
     // Write out the embedded tools and scripts.
     unpack_tarball(&dir)
         .await
         .context("Unable to install tools")?;
 
+    // Pick one of the embedded files for use as the canonical mtime.
```

Comment:
Why do we need a canonical mtime? Is this preventing docker rebuilds?

Reply:
> Why do we need a canonical mtime? Is this preventing docker rebuilds?

I'll add a comment to the code. It's a minor optimization to make the build context more efficient by not always copying the tools through.

For example on a local build with a 180 MiB file:
```
❯ docker build --progress=plain .
...
#10 [internal] load build context
#10 sha256:8fe1ebd2168d7202972f3fed513e650e6bc14c1252ef426d770005b1b769b70b
#10 transferring context: 183.54MB 0.5s done
#10 DONE 0.6s

❯ docker build --progress=plain .
...
#10 [internal] load build context
#10 sha256:942722aa20dadaa1f21a60c65ffa04d801e48ac344603e42b368fda45457366a
#10 transferring context: 291B done
#10 DONE 0.0s

❯ touch -r Dockerfile NVIDIA-Linux-aarch64-470.82.01.run

❯ docker build --progress=plain .
...
#10 [internal] load build context
#10 sha256:8fe1ebd2168d7202972f3fed513e650e6bc14c1252ef426d770005b1b769b70b
#10 transferring context: 183.54MB 0.5s done
#10 DONE 0.6s

❯ docker build --progress=plain .
...
#10 [internal] load build context
#10 sha256:ec46d1475e765e302cc924f8682c05c897c1a56df7a3b356bae7b1b866eb4c74
#10 transferring context: 291B done
#10 DONE 0.0s
```

The build context is more intelligent than a Docker layer in the sense that modifying a single file results in only that file being transferred, not the entire build context. But it still seems worth avoiding unnecessary copies.

The direction I'm exploring with passing file descriptors could let us get rid of all the changes to the build context. `/packages/*/*.tar.*`, `/build/rpms`, `/.gomodcache`, and `/.cargo` are the current offenders. I'd like to avoid others.